src/group_theory/perm/cycles.lean

```diff
@@ -341,30 +352,26 @@ noncomputable def cycle_factors [fintype α] [linear_order α] (f : perm α) :
   {l : list (perm α) // l.prod = f ∧ (∀ g ∈ l, is_cycle g) ∧ l.pairwise disjoint} :=
 cycle_factors_aux (univ.sort (≤)) f (λ _ _, (mem_sort _).2 (mem_univ _))
 
+/-- Factors a permutation `f` into a list of disjoint cyclic permutations that multiply to `f`,
+  without a linear order. -/
+noncomputable def trunc_cycle_factors [fintype α] (f : perm α) :
+  trunc {l : list (perm α) // l.prod = f ∧ (∀ g ∈ l, is_cycle g) ∧ l.pairwise disjoint} :=
+quotient.rec_on_subsingleton (@univ α _).1
+  (λ l h, trunc.mk (cycle_factors_aux l f h))
+  (show ∀ x, f x ≠ x → x ∈ (@univ α _).1, from λ _ _, mem_univ _)
```

Comment:
It seems weird to me to use `trunc` for something that's noncomputable. What's the advantage of `trunc` over `nonempty` in this case?

Reply:
Or alternatively; how hard is it to make `cycle_factors_aux` computable?

Reply:
I have no idea how hard it is to make `cycle_factors_aux` computable, but you're right that I should just make it `nonempty` unless I can figure that out.

Reply:
Ugh, looks like it became non-computable in 6f6b548b2c2f2d54785630b097e474345ccac848

Reply:
It's computable now.